cmd/probe/seachestprobe.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/cmd/controller"
+	"github.com/openebs/node-disk-manager/pkg/seachest"
+	"github.com/openebs/node-disk-manager/pkg/util"
+)
+
+// seachest contains required variables for populating diskInfo
+type seachestProbe struct {
+	// Every new probe needs a controller object to register itself.
+	// Here Controller consists of Clientset, kubeClientset, probes, etc which is used to
+	// create, update, delete, deactivate the disk resources or list the probes already registered.
+	Controller         *controller.Controller
+	SeachestIdentifier *seachest.Identifier
+}
+
+const (
+	seachestConfigKey     = "seachest-probe"
+	seachestProbePriority = 2
+)
+
+var (
+	seachestProbeName  = "seachest probe"
+	seachestProbeState = defaultEnabled
+)
+
+// init is used to get a controller object and then register itself
+var seachestProbeRegister = func() {
+	// Get a controller object
+	ctrl := <-controller.ControllerBroadcastChannel
+	if ctrl == nil {
+		glog.Error("unable to configure", seachestProbeName)
+		return
+	}
+	if ctrl.NDMConfig != nil {
+		for _, probeConfig := range ctrl.NDMConfig.ProbeConfigs {
+			if probeConfig.Key == seachestConfigKey {
+				seachestProbeName = probeConfig.Name
+				seachestProbeState = util.CheckTruthy(probeConfig.State)
+				break
+			}
+		}
+	}
+	newRegisterProbe := &registerProbe{
+		priority:   seachestProbePriority,
+		name:       seachestProbeName,
+		state:      seachestProbeState,
+		pi:         &seachestProbe{Controller: ctrl},
+		controller: ctrl,
+	}
+	// Here we register the probe (smart probe in this case)
```

Comment:
comment need to be fixed '(seachest probe in this case)'

Reply:
Fixed in new PR !!